types/compute-argmax/index.d.ts

```diff
@@ -0,0 +1,8 @@
+// Type definitions for compute-argmax 1.0
+// Project: https://github.com/compute-io/argmax
+// Definitions by: Eric Crosson <https://github.com/EricCrosson>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function argmax(values: number[]): number[];
```

Comment:
Let's simplify `values` type to `ArrayLike<number>`?

```suggestion
declare function argmax(values: ArrayLike<number>): number[];
```

Reply:
Seems like a good idea, thank you.

I'm a little hazy on what may be considered arraylike but is not an array, is there any concrete example you can provide to help my mental model?

Reply:
You're right! Sorry for my carelessness. The [source code checks] to see if it is an array. If you remove the check, you can use it, for example, with `arguments`.

```js
function argmax(arr) {
  var len = arr.length, max = arr[0], idx = [0], val;

  for (var i = 1; i < len; i++) {
    val = arr[i];
    if (val > max) {
      max = val;
      idx.length = 0;
      idx.push(i);
    } else if (val === max) {
      idx.push(i);
    }
  }
  return idx;
}

function test() {
  return argmax(arguments)
}

test(3, 2, 5, 2, 5); // [2, 4]
```

Therefore, I suggest changing this to:

```ts
declare function argmax(values: ReadonlyArray<number>): number[];
```

[source code checks]: https://github.com/compute-io/argmax/blob/master/lib/index.js#L39-L41

Reply:
Oh neato! I always thought `arguments` *was* an array. The [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) say otherwise, thanks for teaching me something today